lib/matplotlib/figure.py

```diff
@@ -1905,6 +1909,16 @@ def _do_layout(gs, layout, unique_ids, nested):
         rows, cols = layout.shape
         gs = self.add_gridspec(rows, cols, **gridspec_kw)
         ret = _do_layout(gs, layout, *_identify_keys_and_nested(layout))
+        ax0 = next(iter(ret.values()))
+        # Only accept strict bools to allow a possible future API expansion.
+        _api.check_isinstance(bool, sharex=sharex, sharey=sharey)
```

Comment:
Does line 1906/1910 modify the figure? If so, I wonder if this check should go before it, so that the figure is not broken (unless it doesn't break)?

Reply:
fair point, moved the check to the top.